edn_format/edn_lex.py

```diff
@@ -174,9 +175,9 @@ def t_WHITESPACE(t):
 
 
 def t_CHAR(t):
-    # from "!" to "~" = all printable ASCII chars except the space
-    r"(\\[!-~])"
-    t.value = t.value[1]
+    # uXXXX hex code or from "!" to "~" = all printable ASCII chars except the space
+    r"(\\u[0-9A-Fa-f]{4}|\\[!-~\w])"
+    t.value = (t.value[1] if len(t.value) == 2 else _bytes(t.value).decode('raw_unicode_escape'))
```

Comment:
Does it work as expected with Python 2, where `len("\\Σ")` is `3` but `len(u"\\Σ")` is `2`?

Reply:
Short answer: yes (I've ran tests for both py2 & py3)
Long answer: the reader treats input as unicode, thus `\w` is applied in unicode mode, and result is also a unicode string (so even if you pass `'\\\xce\xa3'` it will still be read as `u"\\Σ"` and parsed as such)